src/main/java/frc/robot/subsystems/scoring/ScoringSubsystem.java

```diff
@@ -0,0 +1,126 @@
+package frc.robot.subsystems.scoring;
+
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.subsystems.shooter.ShooterIOInputsAutoLogged;
+
+public class ScoringSubsystem extends SubsystemBase {
+    private final ShooterIO shooterIo;
+    private final ShooterIOInputsAutoLogged shooterInputs = new ShooterIOInputsAutoLogged();
+
+    private final AimerIO aimerIo;
+    private final AimerIOInputsAutoLogged aimerInputs = new AimerIOInputsAutoLogged();
+
+    private final Timer timer = new Timer();
+
+    private enum ScoringState {
+        WAIT,
+        INTAKE,
+        PRIME,
+        SHOOT,
+        ENDGAME
+    }
+
+    private ScoringState state = ScoringState.WAIT;
```

Comment:
In addition to this, let's create a enum called ScoringAction to enumerate the scoring actions the user/software can do.

This will allow us to separate user commands from the state machine itself, which will be really helpful for ensuring the state transitions occur as we want